test/test_datadog_agent.cpp

```diff
@@ -66,6 +66,23 @@ TEST_CASE("CollectorResponse") {
     REQUIRE(logger->error_count() == 0);
   }
 
+  SECTION("HTTP success with empty body") {
+    // Don't echo error messages.
+    logger->echo = nullptr;
+
+    {
+      http_client->response_status = 200;
+      http_client->response_body.str("");
+      http_client->response_body.clear();
```

Comment:
This testing library we're using, Catch2, has the neat behavior that each `SECTION` is isolated from its cousins.

That is, in the following:
```
TEST_CASE("foo") {
    a;
    SECTION("bar") {
        b;
    }
    c;
    SECTION("baz") {
        d;
        SECTION("bam") {
            e;
        }
        f;
        SECTION("barf") {
            g;
        }
        h;
    }
    i;
}
```
The `TEST_CASE("foo")` will run three times: once for each "leaf" `SECTION` in the tree of nested `SECTION`s.

The order of statements in each run will be:

1. a, b, c, i
2. a, c, d, e, f, h, i
3. a, c, d, f, g, h, i

Pretty neat, right? Gets out of hand pretty quickly, though, with more than a layer or so of nesting.

That's a long way of saying that `http_client->response_body.clear();` is probably not necessary.

Reply:
That's a really nice and simple introduction de Catch2 I would have loved to have when I encountered it two years ago, it would have saved me so many mistakes. Thank you.

fixed in 8c405b49c68a5c20b5417a1c76261e1e49672b75